### PR TITLE
Don't send comment email to suspended or inactive users.

### PR DESCRIPTION
**Pull request by steveyken**
Users who are suspended, inactive or not yet confirmed should not receive comment notifications.

### CODE REVIEW

**Comment by steveyken** [comment]
I need to fix failing feature spec before merging

File: app/models/polymorphic/comment.rb
Comment (CloCkWeRX) [inline app/models/polymorphic/comment.rb:57]:
```suggestion
      if subscriber&.confirmed? && !subscriber.awaits_approval? && !subscriber.suspended? && subscriber.email.present?
```

File: app/models/polymorphic/comment.rb
Comment (CloCkWeRX) [inline app/models/polymorphic/comment.rb:57]: Can you refactor this to a method on User, something like `emailable?`

File: app/models/polymorphic/comment.rb
Comment (CloCkWeRX) [inline app/models/polymorphic/comment.rb:56]: Assuming this is an activerecord relation; not an array of ids (hard to tell), the below
```
commentable.subscribed_users.reject { |user_id| user_id == user.id }
```
should be refactored to a scope or `.where.not(user+id: user.id)`; so that you don't do a loop of user objects you already have.

If its an array of user ids; it should still be refactored to something like
```
other_users = User.where(id: commentable.subscribed_users.reject { |user_id| user_id == user.id }.collect(&:subscriber_id))
```
Then you can further add scopes for currently_emailable applying your conditions below.

Avoid N+1 issues

**Review by CloCkWeRX — CHANGES_REQUESTED** [review]
N+1 issues if it is likely to be more than 1-2 subscribed users for a comment; .emailable? or a similar method to consolidate the logic in one spot.

**Comment by steveyken** [comment]
Good call, I'll make those changes.

Unfortunately subscribed_users is an array of IDs rather than a relation.
We really need to change that in future. I've popped in an issue just for
that.

**Comment by steveyken** [comment]
@CloCkWeRX ok to merge? let me know if any further thoughts

**Review by CloCkWeRX — APPROVED** [review]

**Pull request merged**
